### PR TITLE
Add support for Kafka Exporter exclude parameters

### LINKED ISSUE #8555: Add support for topic and consumer group exclusion in Kafka Exporter
#8554 updates Kafka Exporter to 1.7.0. This version adds new parameters for topic and consumer group exclusion. It would be good to support their configuration through the `Kafka` custom resource in the `.spec.kafkaExporter` section.

### PR DESCRIPTION
**Pull request by Thealisyed**
### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

This PR fixes #8555 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [X] Write tests
- [X] Make sure all tests pass
- [X] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [X] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [X] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java [hunk 1]
```diff
@@ -36,6 +37,8 @@ public class KafkaExporterSpec implements HasLivenessProbe, HasReadinessProbe, U
     private String image;
     private String groupRegex = ".*";
     private String topicRegex = ".*";
+    private String topicExcludeRegex = "^$";
+    private String groupExcludeRegex = "^$";
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java:41]: Would null make more sense as the default value compared to this?

File: examples/metrics/kafka-metrics.yaml [hunk 2]
```diff
@@ -65,6 +65,8 @@ spec:
   kafkaExporter:
     topicRegex: ".*"
     groupRegex: ".*"
+    topicExcludeRegex: "^$"
+    groupExcludeRegex: "^$"
```
Comment (scholzj) [inline examples/metrics/kafka-metrics.yaml:69]: You should not change anything in `examples/`, `install/` or `helm-charts/`. All changes should be done to `packaging/` only.

File: packaging/examples/metrics/kafka-metrics.yaml [hunk 3]
```diff
@@ -65,6 +65,8 @@ spec:
   kafkaExporter:
     topicRegex: ".*"
     groupRegex: ".*"
+    topicExcludeRegex: "^$"
+    groupExcludeRegex: "^$"
```
Comment (scholzj) [inline packaging/examples/metrics/kafka-metrics.yaml:69]: I'm not sure we want to include these in the examples as they have rather secondary meaning ... what do you think @ppatierno?

File: systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsIsolatedST.java [hunk 4]
```diff
@@ -361,6 +361,8 @@ void testKafkaExporterDifferentSetting() throws InterruptedException, ExecutionE
         KafkaResource.replaceKafkaResourceInSpecificNamespace(kafkaClusterFirstName, k -> {
             k.getSpec().getKafkaExporter().setGroupRegex("my-group.*");
             k.getSpec().getKafkaExporter().setTopicRegex(topicName);
+            k.getSpec().getKafkaExporter().setGroupExcludeRegex("my-group.*");
+            k.getSpec().getKafkaExporter().setTopicExcludeRegex(topicName);
```
Comment (scholzj) [inline systemtest/src/test/java/io/strimzi/systemtest/metrics/MetricsIsolatedST.java:365]: Is this really needed for the test to pass? Same below as well.

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java [hunk 5]
```diff
@@ -77,6 +80,28 @@ public void setTopicRegex(String topicRegex) {
         this.topicRegex = topicRegex;
     }
 
+    @Description("Regular expression to specify which consumer groups to exclude. " +
+            "Default value is `^$`.")
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java:84]: if you are going to get the Jakub's suggestion (which I agree on), remember to update the documentation here as well.

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java [hunk 6]
```diff
@@ -77,6 +80,28 @@ public void setTopicRegex(String topicRegex) {
...
+    public String getGroupExcludeRegex() {
+        return groupExcludeRegex;
+    }
+
+    public void setGroupExcludeRegex(String groupExcludeRegex) {  
+        this.groupExcludeRegex = groupExcludeRegex;
+    }
+
+    @Description("Regular expression to specify which topics to exclude. " +
+            "Default value is `^$`.")
```
Comment (ppatierno) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java:95]: Ditto as above.

File: packaging/examples/metrics/kafka-metrics.yaml (hunk 3 above)
Comment (ppatierno) [reply packaging/examples/metrics/kafka-metrics.yaml:69]: Yes I agree.

**Comment by scholzj** [comment]
It looks like you have to update the CRDs to get the builds running:
```
ERROR: Uncommitted changes in derived resources:
M	packaging/helm-charts/helm3/strimzi-kafka-operator/crds/040-Crd-kafka.yaml
Run the following to add up-to-date resources:
  mvn clean verify -DskipTests -DskipITs \
    && make crd_install \
    && make dashboard_install \
    && make helm_install \
    && git add packaging/install/ packaging/helm-charts/ documentation/modules/appendix_crds.adoc cluster-operator/src/main/resources/cluster-roles \
    && git commit -s -m 'Update derived resources'
```

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java [hunk 7]
```diff
@@ -77,6 +80,28 @@ public void setTopicRegex(String topicRegex) {
         this.topicRegex = topicRegex;
     }
 
+    @Description("Regular expression to specify which consumer groups to exclude. " +
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java:83]: I think that there is simply no default value => so I would remove the `Default value is null.`.

File: api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java [hunk 8]
```diff
@@ -77,6 +80,28 @@ public void setTopicRegex(String topicRegex) {
...
+    @JsonInclude(JsonInclude.Include.NON_DEFAULT)
+    public String getGroupExcludeRegex() {
+        return groupExcludeRegex;
+    }
+
+    public void setGroupExcludeRegex(String groupExcludeRegex) {  
+        this.groupExcludeRegex = groupExcludeRegex;
+    }
+
+    @Description("Regular expression to specify which topics to exclude. " +
```
Comment (scholzj) [inline api/src/main/java/io/strimzi/api/kafka/model/KafkaExporterSpec.java:94]: Same as above ... I think that there is simply no default value => so I would remove the `Default value is null.`.

File: documentation/modules/metrics/proc-metrics-kafka-deploy-options.adoc [hunk 9]
```diff
@@ -123,16 +123,18 @@ spec:
     image: my-registry.io/my-org/my-exporter-cluster:latest <1>
     groupRegex: ".*" <2>
     topicRegex: ".*" <3>
-    resources: <4>
+    groupExcludeRegex: "^$" <4>
+    topicExcludeRegex: "^$" <5>
```
Comment (scholzj) [inline documentation/modules/metrics/proc-metrics-kafka-deploy-options.adoc:127]: Maybe you can add some better examples here? E.g. exclude some topics / groups?

**Review by scholzj — COMMENTED** [review]
Some more nits. Also, don't forget to update the CRDs, so that I can run the STs. Bu otherwise it looks like a nice work.

File: documentation/modules/metrics/proc-metrics-kafka-deploy-options.adoc [hunk 10]
```diff
@@ -123,16 +123,18 @@ spec:
     image: my-registry.io/my-org/my-exporter-cluster:latest <1>
     groupRegex: ".*" <2>
     topicRegex: ".*" <3>
-    resources: <4>
+    groupExcludeRegex: "Exlcude some groups" <4>
+    topicExcludeRegex: "Exclude some topics" <5>
```
Comment (scholzj) [inline documentation/modules/metrics/proc-metrics-kafka-deploy-options.adoc:127]: What about doing something like this to give it an example look?

```suggestion
    groupExcludeRegex: "^excluded-.*" <4>
    topicExcludeRegex: "^excluded-.*" <5>
```

**Review by scholzj — APPROVED** [review]
One nit. LGTM otherwise.

**Pull request merged**